#  Statistics for Decision Making

# Question 1: For the suburb Altona, it is postulated that a typical property sells for 800,000. Use the data at hand to test this assumption. Is the typical property price really 800,000 or has it increased? Use a significance level of 5%.

In [9]:
import pandas as pd
from scipy.stats import ttest_1samp

data = pd.read_csv("/Users/akhilnair/Downloads/property (1).csv")
data

Suburb           Address  Rooms Type      Price Method  \
0         Abbotsford      85 Turner St      2    h  1480000.0      S   
1         Abbotsford   25 Bloomburg St      2    h  1035000.0      S   
2         Abbotsford      5 Charles St      3    h  1465000.0     SP   
3         Abbotsford  40 Federation La      3    h   850000.0     PI   
4         Abbotsford       55a Park St      4    h  1600000.0     VB   
...              ...               ...    ...  ...        ...    ...   
13575  Wheelers Hill      12 Strada Cr      4    h  1245000.0      S   
13576   Williamstown     77 Merrett Dr      3    h  1031000.0     SP   
13577   Williamstown       83 Power St      3    h  1170000.0      S   
13578   Williamstown      96 Verdon St      4    h  2500000.0     PI   
13579     Yarraville        6 Agnes St      4    h  1285000.0     SP   

        SellerG        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  \
0        Biggin   3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0   
1        Biggin   4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0   
2        Biggin   4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0   
3        Biggin   4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0   
4        Nelson   4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0   
...         ...         ...       ...       ...  ...       ...  ...       ...   
13575     Barry  26/08/2017      16.7    3150.0  ...       2.0  2.0     652.0   
13576  Williams  26/08/2017       6.8    3016.0  ...       2.0  2.0     333.0   
13577     Raine  26/08/2017       6.8    3016.0  ...       2.0  4.0     436.0   
13578   Sweeney  26/08/2017       6.8    3016.0  ...       1.0  5.0     866.0   
13579   Village  26/08/2017       6.3    3013.0  ...       1.0  1.0     362.0   

       BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  \
0               NaN        NaN        Yarra -37.79960   144.99840   
1              79.0     1900.0        Yarra -37.80790   144.99340   
2             150.0     1900.0        Yarra -37.80930   144.99440   
3               NaN        NaN        Yarra -37.79690   144.99690   
4             142.0     2014.0        Yarra -37.80720   144.99410   
...             ...        ...          ...       ...         ...   
13575           NaN     1981.0          NaN -37.90562   145.16761   
13576         133.0     1995.0          NaN -37.85927   144.87904   
13577           NaN     1997.0          NaN -37.85274   144.88738   
13578         157.0     1920.0          NaN -37.85908   144.89299   
13579         112.0     1920.0          NaN -37.81188   144.88449   

                       Regionname Propertycount  
0           Northern Metropolitan        4019.0  
1           Northern Metropolitan        4019.0  
2           Northern Metropolitan        4019.0  
3           Northern Metropolitan        4019.0  
4           Northern Metropolitan        4019.0  
...                           ...           ...  
13575  South-Eastern Metropolitan        7392.0  
13576        Western Metropolitan        6380.0  
13577        Western Metropolitan        6380.0  
13578        Western Metropolitan        6380.0  
13579        Western Metropolitan        6543.0  

[13580 rows x 21 columns]

In [10]:
# Filtering the data for the suburb Altona
altona_data = data[data['Suburb'] == 'Altona']
altona_data

Suburb          Address  Rooms Type      Price Method        SellerG  \
141    Altona     158 Queen St      3    h   520000.0     VB           Greg   
142    Altona    59 Bracken Gr      5    h  1525000.0      S           Greg   
143    Altona   1/123 Blyth St      2    t   720000.0      S  hockingstuart   
144    Altona       4 Blyth St      3    h  1120000.0      S         Barlow   
145    Altona     20 Linnet St      4    h   780000.0     PI        Sweeney   
...       ...              ...    ...  ...        ...    ...            ...   
12216  Altona    86 Grieve Pde      3    h   913000.0      S           Greg   
12432  Altona  7/35 Grieve Pde      2    u   458000.0     SP  hockingstuart   
12634  Altona      6 Myrtle Gr      3    h  1248500.0      S             RT   
13338  Altona       20 Burt St      3    h   920000.0     VB            PRD   
13339  Altona    1/76 Queen St      2    u   801250.0      S        Sweeney   

             Date  Distance  Postcode  ...  Bathroom  Car  Landsize  \
141     3/09/2016      13.8    3018.0  ...       2.0  1.0     352.0   
142     3/12/2016      13.8    3018.0  ...       3.0  3.0     729.0   
143     4/02/2016      13.8    3018.0  ...       1.0  2.0     292.0   
144     4/03/2017      13.8    3018.0  ...       1.0  2.0     506.0   
145     4/06/2016      13.8    3018.0  ...       1.0  4.0     655.0   
...           ...       ...       ...  ...       ...  ...       ...   
12216   3/09/2017      11.0    3018.0  ...       2.0  3.0     530.0   
12432   9/09/2017      11.0    3018.0  ...       1.0  1.0      63.0   
12634  16/09/2017      11.0    3018.0  ...       1.0  3.0     710.0   
13338  26/08/2017      11.0    3018.0  ...       2.0  2.0     584.0   
13339  26/08/2017      11.0    3018.0  ...       1.0  1.0     245.0   

       BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  \
141           242.0     2015.0  Hobsons Bay -37.87000   144.82500   
142           268.0     2005.0  Hobsons Bay -37.87210   144.81050   
143           125.0     2013.0  Hobsons Bay -37.86870   144.81970   
144           120.0     1930.0  Hobsons Bay -37.86930   144.83750   
145             NaN        NaN  Hobsons Bay -37.86800   144.81540   
...             ...        ...          ...       ...         ...   
12216         148.0     1985.0          NaN -37.86140   144.81931   
12432          75.0     1975.0          NaN -37.86933   144.81704   
12634           NaN        NaN          NaN -37.87182   144.81600   
13338           NaN        NaN          NaN -37.86249   144.82162   
13339           NaN        NaN          NaN -37.87046   144.83482   

                 Regionname Propertycount  
141    Western Metropolitan        5301.0  
142    Western Metropolitan        5301.0  
143    Western Metropolitan        5301.0  
144    Western Metropolitan        5301.0  
145    Western Metropolitan        5301.0  
...                     ...           ...  
12216  Western Metropolitan        5301.0  
12432  Western Metropolitan        5301.0  
12634  Western Metropolitan        5301.0  
13338  Western Metropolitan        5301.0  
13339  Western Metropolitan        5301.0  

[74 rows x 21 columns]

In [11]:
# Removing rows with missing price values
altona_data = altona_data.dropna(subset=['Price'])
# Performing a one-sample t-test to compare the mean property price in Altona to $800,000
# Null hypothesis: The typical property price in Altona is $800,000
# Alternative hypothesis: The typical property price in Altona is not $800,000

population_mean = 800000

# Performing the t-test
t_stat, p_value = ttest_1samp(altona_data['Price'], population_mean)

# Determine if we reject the null hypothesis (Significance Level=alpha)
alpha = 0.05
reject_null = p_value < alpha

t_stat, p_value, reject_null

(1.0277020770199676, 0.3074832713055549, False)

Since the p-value (0.307) is greater than the significance level of 0.05, we fail to reject the null hypothesis. This means there is not enough evidence to conclude that the typical property price in Altona is different from $800,000. Based on this data, we cannot say that the typical property price has increased significantly.

# Question 2:  For the year 2016, is there any difference in prices of properties sold in the summer months vs winter months? Consider months from October till March as winter months and rest as summer months. Use a significance level of 5%. 

In [12]:
from scipy.stats import ttest_ind

# Filtering the data for the year 2016
data_2016 = data[data['Date'].str.contains('2016')]

# Converting the Date column to datetime format
data_2016['Date'] = pd.to_datetime(data_2016['Date'], format='%d/%m/%Y')

# Defining summer and winter months
winter_months = [10, 11, 12, 1, 2, 3]
summer_months = [4, 5, 6, 7, 8, 9]

# Splitting the data into winter and summer months
winter_data = data_2016[data_2016['Date'].dt.month.isin(winter_months)]
summer_data = data_2016[data_2016['Date'].dt.month.isin(summer_months)]

# Removing rows with missing price values
winter_data = winter_data.dropna(subset=['Price'])
summer_data = summer_data.dropna(subset=['Price'])

# Performing an independent t-test to compare the mean property prices between winter and summer months
t_stat, p_value = ttest_ind(winter_data['Price'], summer_data['Price'])

# Determine if we reject the null hypothesis (Significance Level=alpha
alpha = 0.05
reject_null = p_value < alpha

t_stat, p_value, reject_null

/var/folders/7q/tjdk40tn4g58qrl_wg3r9b700000gn/T/ipykernel_9627/3505397000.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_2016['Date'] = pd.to_datetime(data_2016['Date'], format='%d/%m/%Y')


(4.043386317851058, 5.330976766763168e-05, True)

Since the p-value (0.000053) is less than the significance level of 0.05, we reject the null hypothesis. This means there is a statistically significant difference in property prices between the summer months and winter months in the year 2016.

# Question 3: For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have car parking? Use the column car in the dataset. Round off your answer to 3 decimal places. 

In [13]:
from scipy.stats import binom

# Load the data
data = pd.read_csv("/Users/akhilnair/Downloads/property (1).csv")

# Filtering the data for the suburb Abbotsford
abbotsford_data = data[data['Suburb'] == 'Abbotsford']

# Calculating the probability of a property not having car parking
no_car_parking_prob = (abbotsford_data['Car'] == 0).mean()

# Calculating the probability that out of 10 properties sold, 3 will not have car parking
n = 10
k = 3
probability = binom.pmf(k, n, no_car_parking_prob)

# Rounding off the result to 3 decimal places
probability_rounded = round(probability, 3)

no_car_parking_prob, probability_rounded


(0.26785714285714285, 0.26)

The probability that out of 10 properties sold in the suburb of Abbotsford, 3 will not have car parking is approximately 0.26.

# Question 4: In the suburb Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places. 

In [14]:
# Calculating the probability of finding a property with 3 rooms in Abbotsford
three_rooms_prob = (abbotsford_data['Rooms'] == 3).mean()

# Rounding off the result to 3 decimal places
three_rooms_prob_rounded = round(three_rooms_prob, 3)

three_rooms_prob_rounded

0.357

The probability of finding a property with 3 rooms in the suburb of Abbotsford is approximately 0.357.

# Question 5: In the suburb Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places.

In [15]:
# Calculating the probability of finding a property with 2 bathrooms in Abbotsford
two_bathrooms_prob = (abbotsford_data['Bathroom'] == 2).mean()

# Rounding off the result to 3 decimal places
two_bathrooms_prob_rounded = round(two_bathrooms_prob, 3)

two_bathrooms_prob_rounded

0.339

The probability of finding a property with 2 bathrooms in the suburb of Abbotsford is approximately 0.339.